### PR TITLE
feat: configure kubo from custom container

### LINKED ISSUE #9: Configure IPFS node
For this we will need to provide ipfs as a custom dockerfile, same as we do for the pickup worker image here https://github.com/olizilla/pickup/blob/65a1d9b1b20201119f80a9fe659455b394091e7a/stacks/PickupStack.ts#L14

An example of setting up your ipfs node from a container init script is here
https://github.com/ipfs-shipyard/go-ipfs-docker-examples/blob/main/gateway/Dockerfile

Some config changes to make
```bash
# dont add provider records to the dht
ipfs config --json Experimental.StrategicProviding true

# maybe have go faster dht... but makes it unusable for the first 5 mins!! https://github.com/ipfs/kubo/blob/master/docs/experimental-features.md#accelerated-dht-client
ipfs config --json Experimental.AcceleratedDHTClient true

# as per gateways
ipfs config --json Swarm.DisableBandwidthMetrics true

# no MDNS plz
ipfs config --json Discovery.MDNS.Enabled false

# plz fail early if bits get flipped in blockstore
ipfs config --json Datastore.HashOnRead true


```

if we want auto GC-ing then... but might be better to just do it manually.
```
# Gotta GC, set with reference to the ephemeral storage available
ipfs config --json Datastore.StorageMax "200GB"
ipfs config --json Datastore.StorageGCWatermark "50"
ipfs config --json Datastore."GCPeriod" "10m"
# Maybe manually GC after each upload as well.
```

### LINKED ISSUE #21: Check if elastic-ipfs already has a CID before asking kubo
If a CID is provided by elastic-ipfs then we should respond that the CID is already pinned. We should not ask kubo to try and fetch it.

Try using dagular to fetch the root CID from elastic-ipfs first. We just want a has check against a known peer.

### PR DESCRIPTION
**Pull request by olizilla**
- Adds a dockerfile to init kubo with customised config.
- Forces platform to amd64 for images intended to run in ECS. Without this, the images built locally are platform dependant, and will fail if built from, say, and an `arm64` machine. ~~I can't see a way to configure aws-cdk to set the platform when building, so we have to force it in the docker file. Thanks to https://stackoverflow.com/a/70614305/6490163~~
  - Updating sst and aws-sdk-lib so we can set the platform per https://github.com/aws/aws-cdk/pull/20563

fixes #9
fixes #21

License: MIT
Signed-off-by: Oli Evans <oli@protocol.ai>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # I build the pickup worker that subscribes to the SQS Queue.
-FROM node:16-alpine
+FROM --platform=linux/amd64 node:16-alpine
```
Comment (olizilla) [inline Dockerfile:2]: The real fix here would be to teach aws-cdk to let us specify the platform build arg when it runs the build but that is  a harder path.

**Comment by olizilla** [comment]
Oh my https://github.com/aws/aws-cdk/issues/12472

**Comment by olizilla** [comment]
So the real fix exists and was merged in this PR https://github.com/aws/aws-cdk/pull/20563 but we SST requires we stick to a specific version of the aws-cdk for things to work, so we need to wait for an SST update to get it.

**Review by alanshaw — APPROVED** [review]

**Comment by olizilla** [comment]
```console
$ npx sst update v1.7.0
Updating @serverless-stack/cli to v1.7.0
Updating @serverless-stack/resources to v1.7.0
Updating aws-cdk-lib@2.32.0
SST: 1.7.0
CDK: 2.32.0

```

**Pull request merged**